packages/react/src/auto/mui/inputs/MUIAutoFileInput.tsx

```diff
@@ -0,0 +1,66 @@
+import { Button, styled } from "@mui/material";
+import React, { useCallback } from "react";
+import type { Control } from "react-hook-form";
+import { useController } from "react-hook-form";
+import { useApi } from "../../../GadgetProvider.js";
+import { useFieldMetadata } from "../../hooks/useFieldMetadata.js";
+
+export interface MUIFileInputProps {
+  label?: string;
+  value?: File;
+  onChange: (value: File) => void;
+}
+const VisuallyHiddenInput = styled("input")({
+  clip: "rect(0 0 0 0)",
+  clipPath: "inset(50%)",
+  height: 1,
+  overflow: "hidden",
+  position: "absolute",
+  bottom: 0,
+  left: 0,
+  whiteSpace: "nowrap",
+  width: 1,
+});
+
+export const MUIAutoFileInput = (props: { field: string; control?: Control<any> }) => {
+  const { field: fieldApiIdentifier, control } = props;
+  const { path, metadata } = useFieldMetadata(fieldApiIdentifier);
+  const api = useApi();
+
+  const { field: fieldProps } = useController({
+    control,
+    name: path,
+  });
+
+  const uploadFile = useCallback(
+    async (files: File[]) => {
+      const { url, token } = await (api as any).getDirectUploadToken();
+      await fetch(url, {
+        method: "PUT",
+        headers: {
+          "Content-Type": files[0].type,
+        },
+        body: files[0],
+      });
+      fieldProps.onChange({
+        directUploadToken: token,
+        fileName: files[0].name,
+      });
+    },
+    [api, fieldProps]
+  );
```

Comment:
`useFieldMetadata, useApi, useController, fetch->fieldProps.onChange(..)` calls are basically the same between the MUI and Polaris variants. Could you move them to a common hook, perhaps something like`useFileInputController` which returns these common functionalities